apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -208,6 +214,62 @@ type BMCDetails struct {
 	DisableCertificateVerification bool `json:"disableCertificateVerification,omitempty"`
 }
 
+// HardwareRAIDVolume defines the desired configuration of volume in hardware RAID
+type HardwareRAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	// +kubebuilder:validation:Minimum=0
+	SizeGibibytes *int `json:"sizeGibibytes,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0;1;1+0.
+	// +kubebuilder:validation:Enum="0";"1";"1+0"
+	Level string `json:"level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	// +kubebuilder:validation:MaxLength=64
+	Name string `json:"name,omitempty"`
+
+	// Select disks with only rotational or solid-state storage
+	Rotational *bool `json:"rotational,omitempty"`
+
+	// Integer, number of disks to use for the logical disk. Defaults to minimum number of disks required
+	// for the particular RAID level.
+	// +kubebuilder:validation:Minimum=1
+	NumberOfPhysicalDisks *int `json:"numberOfPhysicalDisks,omitempty"`
+}
+
+// SoftwareRAIDVolume defines the desired configuration of volume in software RAID
+type SoftwareRAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	// +kubebuilder:validation:Minimum=0
+	SizeGibibytes *int `json:"sizeGibibytes,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0;1;1+0.
+	// +kubebuilder:validation:Enum="0";"1";"1+0"
```

Comment:
Please remind me, we made a conscious choice to exclude 5 and 6?

Reply:
Still needs answering

Reply:
Sorry, I forget it. Here is the part left over before I took over. The reason seems to be because of this [comment](https://github.com/metal3-io/baremetal-operator/pull/292#discussion_r322587719).

Reply:
Let's probably open another patch and discuss there.